diff_cover/violations_reporter.py

```diff
@@ -116,14 +116,14 @@ def _get_classes(self, xml_document, src_path):
              src_abs_path in [
                  os.path.join(
                      source,
-                     clazz.get('filename')
+                     os.path.normpath(clazz.get('filename').replace("\\", "/"))
                  ) for source in sources]]
             or
             [clazz for clazz in classes if
-             clazz.get('filename') == src_abs_path]
+             os.path.normpath(clazz.get('filename').replace("\\", "/")) == src_abs_path]
             or
             [clazz for clazz in classes if
-             clazz.get('filename') == src_rel_path]
+             os.path.normpath(clazz.get('filename').replace("\\", "/")) == src_rel_path]
```

Comment:
`os.path.normpath` will normalise the strings for the machine you're running on, which is why it would have appeared to do nothing for you. Backslashes are a valid character in Unix-style paths, so `.\src\maths.lua` is actually a valid, "normalised" file name. Just converting the paths with `.replace("\\", "/")` should work as intended unless someone actually puts a backslash in one of their paths.